security/pkg/pki/ca/ca.go

```diff
@@ -81,45 +78,22 @@ type IstioCAOptions struct {
 
 	KeyCertBundle util.KeyCertBundle
 
-	LivenessProbeOptions *probe.Options
-	ProbeCheckInterval   time.Duration
-
 	// Config for creating self-signed root cert rotator.
 	RotatorConfig *SelfSignedCARootCertRotatorConfig
 }
 
 // NewSelfSignedIstioCAOptions returns a new IstioCAOptions instance using self-signed certificate.
-func NewSelfSignedIstioCAOptions(ctx context.Context,
-	rootCertGracePeriodPercentile int, caCertTTL, rootCertCheckInverval, defaultCertTTL,
+func NewSelfSignedIstioCAOptions(
+	rootCertGracePeriodPercentile int, caCertTTL, rootCertCheckInterval, defaultCertTTL,
 	maxCertTTL time.Duration, org string, dualUse bool, namespace string,
-	readCertRetryInterval time.Duration, client corev1.CoreV1Interface,
```

Comment:
Why was this removed? Is there no scenario where Istiod needs to wait for the caCerts to be loaded any longer?

Reply:
Currently no, because we donot have the wait until logic anymore